src/linker/Linker/Driver.cs

```diff
@@ -715,6 +723,29 @@ public bool Run (ILogger customLogger = null)
 
 		partial void PreProcessPipeline (Pipeline pipeline);
 
+		private static HashSet<uint> ParseWarnings (string value)
+		{
+			string Unquote (string arg)
+			{
+				if (arg.Length > 1 && arg[0] == '"' && arg[arg.Length - 1] == '"')
+					return arg.Substring (1, arg.Length - 2);
+
+				return arg;
+			}
+
+			value = Unquote (value);
+			string[] values = value.Split (new char[] { ',', ';', ' ' }, StringSplitOptions.RemoveEmptyEntries);
+			HashSet<uint> noWarnCodes = new HashSet<uint> ();
+			foreach (string id in values) {
+				var warningCode = id.StartsWith ("IL") ? id.Substring (2) : id;
```

Comment:
Linker must ONLY recognize prefixed warning codes - so `IL1234` is recognized as linker warning. Everything else in the nowarn must be ignored and linker should not fail on it:
* The NoWarn property is shared between multiple tools, CSC being the most prominent one. It will contain stuff for multiple tools - we must filter out the ones intended for linker and ignore the rest
* For historical reasons, unprefixed numbers in NoWarn are recognized as CS warnings, linker should not "take over" these in any way

There's a separate discussion if this logic should be in the linker or the ILLink task as both could handle this. (Technically it could also be in ILLink target, but it's difficult to implement the logic in MSBuild).

I personally would prefer we don't do "preprocessing" of values in the ILLink task - so I would prefer we keep this in the linker.